internal/testing/testdata/testdata.go

```diff
@@ -1114,9 +1150,193 @@ var (
 	}
 
 	CdxQuarkusIngestionPredicates = assembler.IngestPredicates{
-		IsDependency: CdxQuarkusDeps,
-		IsOccurrence: CdxQuarkusOccurrence,
-		HasSBOM:      CdxQuarkusHasSBOM,
+		IsDependency: cdxQuarkusDeps,
+		IsOccurrence: cdxQuarkusOccurrence,
+		HasSBOM:      cdxQuarkusHasSBOM,
+		CertifyLegal: cdxQuarkusLegal,
+	}
+
+	cdxLegalDeps = []assembler.IsDependencyIngest{
+		{
+			Pkg:             cdxTopQuarkusPack,
+			DepPkg:          cdxResteasyPack,
+			DepPkgMatchFlag: model.MatchFlags{Pkg: model.PkgMatchTypeSpecificVersion},
+			IsDependency: &model.IsDependencyInputSpec{
+				DependencyType: model.DependencyTypeDirect,
+				VersionRange:   "2.13.4.Final",
+				Justification:  isCDXDepJustifyDependsJustification,
+			},
+		},
+		{
+			Pkg:             cdxTopQuarkusPack,
+			DepPkg:          cdxReactiveCommonPack,
+			DepPkgMatchFlag: model.MatchFlags{Pkg: model.PkgMatchTypeSpecificVersion},
+			IsDependency: &model.IsDependencyInputSpec{
+				DependencyType: model.DependencyTypeIndirect,
+				VersionRange:   "2.13.4.Final",
+				Justification:  isCDXDepJustifyDependsJustification,
+			},
+		},
+		{
+			Pkg:             cdxResteasyPack,
+			DepPkg:          cdxReactiveCommonPack,
+			DepPkgMatchFlag: model.MatchFlags{Pkg: model.PkgMatchTypeSpecificVersion},
+			IsDependency: &model.IsDependencyInputSpec{
+				DependencyType: model.DependencyTypeDirect,
+				VersionRange:   "2.13.4.Final",
+				Justification:  isCDXDepJustifyDependsJustification,
+			},
+		},
+		{
+			Pkg:             cdxTopQuarkusPack,
+			DepPkg:          cdxSmallRye,
+			DepPkgMatchFlag: model.MatchFlags{Pkg: model.PkgMatchTypeSpecificVersion},
+			IsDependency: &model.IsDependencyInputSpec{
+				DependencyType: model.DependencyTypeUnknown,
+				VersionRange:   "2.27.0",
+				Justification:  isDepJustifyTopPkgJustification,
+			},
+		},
+	}
+	cdxLegalCertifyLegal = []assembler.CertifyLegalIngest{
+		{
+			Pkg: cdxResteasyPack,
+			Declared: []model.LicenseInputSpec{
+				{
+					Name:        "Apache-2.0",
+					ListVersion: &lvUnknown,
+				},
+			},
+			CertifyLegal: &model.CertifyLegalInputSpec{
+				DeclaredLicense: "Apache-2.0",
+				Justification:   "Found in CycloneDX document",
+				TimeScanned:     cdxQuarkusTime,
+			},
+		},
+		{
+			Pkg: cdxReactiveCommonPack,
+			Declared: []model.LicenseInputSpec{
+				{
+					Name:        "Apache-2.0",
+					ListVersion: &lvUnknown,
+				},
+				{
+					Name:        "LGPL-3.0-or-later",
+					ListVersion: &lvUnknown,
```

Comment:
```
Name: "LicenseRef-asdf1234"
Inline: ptrfrom.String("This is the text of the custom license I wrote")
```
Use common.HashLicense to come up with the asdf1234 part
https://github.com/guacsec/guac/blob/main/pkg/ingestor/parser/common/license.go#L90

Reply:
done